### ISSUE TITLE
NSClient boot.ini functionality or specify remote config during install?

### ISSUE THREAD

**Issue by deloid** [ConSol-Monitoring/snclient#149]
Few feature requests for your consideration, apologies if it's not really an 'issue' as such. I'm currently using a remote https nsclient.ini as the primary config for my agents, but I see similar functionality isn't available in SNClient+ (that I can find). The remote source works well, as any updates to config (wrapped scripts mainly) are very quickly picked up by agents.

Are there any plans to implement settings store configuration like NSClient's boot.ini to specify the config source during service start or the ability to change config sources via snclient.exe?

An installer option to specify a configuration source (ie. remote https) during installation would also be good. On Windows, it's the CONFIGURATION_TYPE property in the  NSClient MSI - https://nsclient.org/docs/installing/#msi-options.

Further to using a remote config as the primary source, a periodic background config reload if the config has changed (hash compare) would also be great - https://nsclient.org/docs/settings/#http-settings.  
I'm coming up blank in trying to find the settings reference, but I believe the periodic reload is handled by this:

```
[/settings/core]
settings maintenance interval=5m
```
I could be way off with that though.

**Comment by sni** [comment]
right now there is no automatic reload of (remote) config includes. But there is a rest api endpoint to trigger a config reload. Would that help?

**Comment by deloid** [comment]
Yes, that does look like it will help in the meantime paired with a scheduled task or something. Thank you for the reply :)

Is a regular hash-check/similar and reload something that could be looked for a future release? It would very useful for on going configuration maintenance. In our case, we add external check script definitions to a remote config fairly regularly.

**Comment by sni** [comment]
right now it's not on the roadmap. But if someone wants to look into this...
